From 3fd4d9559b1bae6eaced62ea83dc22dbd4c4df97 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Tue, 17 Feb 2026 04:11:12 +0000
Subject: [PATCH] fix: Disable Firefox Suggest by default, remove unwanted
 Firefox Suggest UI elements, and ensure that Fsuggestions from Firefox
 Suggest are only queried if suggestions are actually enabled by the user

Signed-off-by: celenity <celenity@celenity.dev>
---
 .../java/org/mozilla/fenix/FenixApplication.kt     |  4 ++--
 .../SearchSuggestionsProvidersBuilder.kt           | 14 +++++++-------
 .../fenix/settings/search/SearchEngineFragment.kt  |  2 +-
 .../main/java/org/mozilla/fenix/utils/Settings.kt  |  6 +++---
 4 files changed, 13 insertions(+), 13 deletions(-)

diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
index 725fe78ce0cc..257bbd6ec406 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt
@@ -426,7 +426,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                         // new search suggestions. The worker requires us to have called
                         // `GlobalFxSuggestDependencyProvider.initialize`, which we did before
                         // scheduling these tasks. When disabled we stop the periodic work.
-                        if (settings().enableFxSuggest) {
+                        if (settings().enableFxSuggest && settings().showNonSponsoredSuggestions) {
                             components.fxSuggest.ingestionScheduler.startPeriodicIngestion()
                         } else {
                             components.fxSuggest.ingestionScheduler.stopPeriodicIngestion()
@@ -515,7 +515,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         queueStorageMaintenance()
         queueNimbusFetchInForeground()
         queueDownloadWallpapers()
-        if (settings().enableFxSuggest) {
+        if (settings().enableFxSuggest && settings().showNonSponsoredSuggestions) {
             queueSuggestIngest()
         }
         queueCollectProcessExitInfo()
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
index 7c68a88f9f45..1ef39c9efaec 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/SearchSuggestionsProvidersBuilder.kt
@@ -78,7 +78,7 @@ class SearchSuggestionsProvidersBuilder(
                 components.core.icons,
                 engineForSpeculativeConnects,
                 showEditSuggestion = false,
-                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             )
 
         defaultCombinedHistoryProvider =
@@ -90,7 +90,7 @@ class SearchSuggestionsProvidersBuilder(
                 engine = engineForSpeculativeConnects,
                 maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                 showEditSuggestion = false,
-                suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+ //               suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             )
 
         val searchBitmap = suggestionIconProvider.getSearchIconBitmap()
@@ -301,7 +301,7 @@ class SearchSuggestionsProvidersBuilder(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -316,7 +316,7 @@ class SearchSuggestionsProvidersBuilder(
                     engine = engineForSpeculativeConnects,
                     maxNumberOfSuggestions = METADATA_SUGGESTION_LIMIT,
                     showEditSuggestion = false,
-                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//                    suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
                     resultsUriFilter = filter::shouldIncludeUri,
                 )
             } else {
@@ -428,7 +428,7 @@ class SearchSuggestionsProvidersBuilder(
                 suggestionIconProvider.getMobileIconDrawable(),
                 suggestionIconProvider.getTabletIconDrawable(),
             ),
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             resultsUrlFilter = filter?.let { it::shouldIncludeUrl },
         )
     }
@@ -450,7 +450,7 @@ class SearchSuggestionsProvidersBuilder(
             components.core.icons,
             suggestionIconProvider.getLocalTabIconDrawable(),
             excludeSelectedSession = !includeSelectedTab,
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             switchToTabDescription = suggestionsStringsProvider.getSwitchToTabDescriptionString(),
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
@@ -474,7 +474,7 @@ class SearchSuggestionsProvidersBuilder(
             indicatorIcon = suggestionIconProvider.getBookmarkIconDrawable(),
             engine = engineForSpeculativeConnects,
             showEditSuggestion = false,
-            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
+//            suggestionsHeader = suggestionsStringsProvider.firefoxSuggestHeader,
             resultsUriFilter = filter?.let { it::shouldIncludeUri },
         )
     }
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
index c0fb12426e4d..dd229dcd1875 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt
@@ -39,7 +39,7 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
         )
 
         requirePreference<SwitchPreference>(R.string.pref_key_show_sponsored_suggestions).apply {
-            isVisible = context.settings().enableFxSuggest
+            isVisible = false
         }
         requirePreference<SwitchPreference>(R.string.pref_key_show_nonsponsored_suggestions).apply {
             isVisible = context.settings().enableFxSuggest
diff --git a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
index 81d95a9666a1..00245c4cf04b 100644
--- a/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
+++ b/mobile/android/fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt
@@ -2489,7 +2489,7 @@ class Settings(
      */
     var enableFxSuggest by lazyFeatureFlagBooleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_enable_fxsuggest),
-        defaultValue = { FxNimbus.features.fxSuggest.value().enabled },
+        defaultValue = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
@@ -2514,7 +2514,7 @@ class Settings(
     var showSponsoredSuggestions by lazyFeatureFlagBooleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_sponsored_suggestions),
         defaultValue = { enableFxSuggest },
-        featureFlag = FeatureFlags.FX_SUGGEST,
+        featureFlag = false,
     )
 
     /**
@@ -2524,7 +2524,7 @@ class Settings(
      */
     var showNonSponsoredSuggestions by lazyFeatureFlagBooleanPreference(
         key = appContext.getPreferenceKey(R.string.pref_key_show_nonsponsored_suggestions),
-        defaultValue = { enableFxSuggest },
+        defaultValue = { false },
         featureFlag = FeatureFlags.FX_SUGGEST,
     )
 
-- 
2.53.0

